### PR TITLE
Store tempest results on devstack jobs

### PR DESCRIPTION
**Pull request by aojea**
It's much better for debugging to store the tempest results in html. 
Also, it stores the tests in junit format, leveraging the Junit plugin in tempest to show the test results, this way you can see the exact number of tests failing from the job itself.

### CODE REVIEW

File: scripts/jenkins/qa_devstack.sh [hunk 1]
```diff
@@ -228,11 +228,13 @@ if [ -e "/dev/vdb" ]; then
     h_setup_extra_disk
 fi
 h_setup_devstack
+
```
Comment (cmurphy) [inline scripts/jenkins/qa_devstack.sh]: (nit) unnecessary whitespace addition

File: scripts/jenkins/qa_devstack.sh [hunk 2]
```diff
@@ -228,11 +228,13 @@ if [ -e "/dev/vdb" ]; then
...
 h_echo_header "Run devstack"
 sudo -u stack -i <<EOF
 cd $DEVSTACK_DIR
 FORCE=yes ./stack.sh
 EOF
+
```
Comment (cmurphy) [inline scripts/jenkins/qa_devstack.sh]: (nit) unnecessary whitespace addition

File: scripts/jenkins/qa_devstack.sh [hunk 3]
```diff
@@ -242,4 +244,14 @@ tox -e smoke
 EOF
 fi
 
+h_echo_header "Store tempest results"
+pip install junitxml
```
Comment (cmurphy) [inline scripts/jenkins/qa_devstack.sh]: I would install the python-junitxml rpm instead

File: scripts/jenkins/qa_devstack.sh [hunk 4]
```diff
@@ -242,4 +244,14 @@ tox -e smoke
...
+sudo -u stack -i <<EOF
+cd /opt/stack
+if [ -f devstack.subunit ]; then
+    subunit2html devstack.subunit
```
Comment (cmurphy) [inline scripts/jenkins/qa_devstack.sh]: Does this need to be output somewhere?

File: jenkins/ci.opensuse.org/openstack-devstack.yaml [hunk 5]
```diff
@@ -53,6 +53,17 @@
               du $file
               exit 1
           fi
+          scp root@cleanvm:/opt/stack/results.html .
+          scp root@cleanvm:/opt/stack/results.xml .
```
Comment (cmurphy) [inline jenkins/ci.opensuse.org/openstack-devstack.yaml]: Maybe use a special directory like .artifacts instead of cluttering up the workspace

Comment (aojea) [reply jenkins/ci.opensuse.org/openstack-devstack.yaml]: I used to store the workspaces because they were cleaned automatically, are we wiping it?

File: scripts/jenkins/qa_devstack.sh (hunk 4 above)
Comment (aojea) [reply scripts/jenkins/qa_devstack.sh]: I couldn't find any documentation but it creates a file in the current folder with the name results.html automatically

File: scripts/jenkins/qa_devstack.sh (hunk 3 above)
Comment (aojea) [reply scripts/jenkins/qa_devstack.sh]: IIRC devstack install everything with pip, and installing with pip works out of the box.
I tried to install the rpm and is giving me dependencies errors, if possible I'd keep to continue using pip.

File: jenkins/ci.opensuse.org/openstack-devstack.yaml (hunk 5 above)
Comment (cmurphy) [reply jenkins/ci.opensuse.org/openstack-devstack.yaml]: I'm not 100% sure the workspace is wiped, it might depend on the job

File: scripts/jenkins/qa_devstack.sh (hunk 4 above)
Comment (cmurphy) [reply scripts/jenkins/qa_devstack.sh]: :+1:

File: scripts/jenkins/qa_devstack.sh (hunk 3 above)
Comment (cmurphy) [reply scripts/jenkins/qa_devstack.sh]: :+1:
